accumulator/transform.go

```diff
@@ -64,6 +64,42 @@ func remTrans2(dels []uint64, numLeaves uint64, forestRows uint8) [][]arrow {
 	return swaps
 }
 
+// makeCollapse determines whether a collapse should take place given a list
+// of nodes to delete and misc info. If the collapse is present, it consists
+// of an single arrow (a source index and a destination index).
+// A collapse is a deferred swap, which is later made into an actual swap in
+// swapCollapses.
+//
+// delRemains must be true if dels has an odd length.
+// r is the row number to generate a collapse for, 0 is the leaf level.
+// rootPresent must be true if there is a tree root in the row r.
+// nextNumLeaves is the final amount of leaves, after the whole deletion
+// process. Note that if r!=0, some dels may already have happened, the slice
+// will be shorter.
+//
+// Example using 15-leaf tree in printout.txt line 21: (forestRows=4, numLeaves=15)
+//
+// Let's say we are deleting nodes 4 through 9 (6 nodes, so delRemains=True).
```

Comment:
delRemains is to grab the leftover 1 node in the tree, so if we're deleting 6 leaves, ```makeCollapse``` should match case default since there is no root in the tree and no delRemains.

Transform is outlined in 7.1 Batch deletion process in the utreexo paper for reference.

Reply:
what do you mean "there is no root"? there is a root on row 0, which is the one the example uses. the root is 14.

Reply:
Ah right yeah there is a root. Oops my bad